api/src/github-user/types.ts

```diff
@@ -0,0 +1,9 @@
+import {} from "module";
+import { GeneralResponseDto } from "../app/types";
+import { GitHubUserApiResponse } from "../github/types";
+import { ValidateNested } from "class-validator";
+
+export class GetUserResponseDto extends GeneralResponseDto {
+  @ValidateNested({ each: true })
+  user?: GitHubUserApiResponse;
```

Comment:
as mentioned in my first comment, use needs to be a dto class in order to show up on swagger docs

```suggestion
  user?: GithubUserDto;
```